### PR TITLE
Append Disc 2 to Titles for Multi-Disc Games

### PR DESCRIPTION
**Pull request by RisingFog**
Fixes issue 5799

https://code.google.com/p/dolphin-emu/issues/detail?id=5799

### CODE REVIEW

**Comment by Mullin** [comment]
I think it's better with multilanguages.

**Comment by JMC47** [comment]
This is an improvement in behavior.

**Comment by lioncash** [comment]
@Mullin is correct that this is a step back with regards to internationalization however. "Disc 2" is not the same way it would be represented in Japanese for example (it would be ディスク2)

**Comment by RisingFog** [comment]
Do not merge yet, needs testing still

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 1]
```diff
@@ -481,7 +481,10 @@ void CGameListCtrl::InsertItemInReportView(long _Index)
 		}
 	}
 
-	SetItem(_Index, COLUMN_TITLE, StrToWxStr(name), -1);
+	if (rISOFile.IsDiscTwo())
+		name = wxString::Format(_("%s (Disc 2)"), name.c_str());;
```
Comment (lioncash) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: Double semicolon

**Comment by Buddybenj** [comment]
Could you also add "(Disc 1)" to Disc 1? Or is there no easy way to do that?

**Comment by RisingFog** [comment]
@Buddybenj I'm not sure if there is an easy way to check for that without having all single disc games having Disc 1

**Comment by skidau** [comment]
How about adding a Disc (number) column to the game list?  The reason being that (Disc X) is not part of the game name and might be misleading.

**Comment by RisingFog** [comment]
@skidau I think it stands out more if it's in the game name, instead of
having it in a very small column which might be obstructed depending on the
size of the Dolphin window.

**Comment by lioncash** [comment]
I may be mistaken, but now that I think of it,  I think some two disc games include the disc number as part of the actual name (I think Baten Kaitos does this) and some don't. I'd have to check when I get home later today.

**Comment by RisingFog** [comment]
@lioncash Resident Evil 4 does do Disc 1 and Disc 2 in their game titles. Perhaps I can do a check to see if Disc 2 is found in the title.

**Comment by tony971** [comment]
End user here - Why not just apply a "Disk 1" to any games that have a "Disk 2" in the user's game catalog?

**Comment by RisingFog** [comment]
@tony971 https://github.com/dolphin-emu/dolphin/pull/2006#issuecomment-72560327

**Comment by tony971** [comment]
Right, I'm proposing that check. If the disk fails "if (rISOFile.IsDiscTwo())" but has a matching title that passes, add "Disk 1." But since I don't know code, feel free to ignore me.

**Comment by RisingFog** [comment]
@tony971 I'll see what can be done about it

**Comment by tueidj** [comment]
@tony971 Because then getting the title for one single game requires scanning the titles of every other game present. Such an exponential increase in processing just for a minor UI enhancement should not be acceptable IMO.

**Comment by RisingFog** [comment]
@tueidj that's what I ended up thinking as well

**Comment by tueidj** [comment]
Also FWIW 4 out of 12 of the two-disc games in my library include the disc number in the disc name.

**Comment by tony971** [comment]
What if the existstance of a disk 2 started the query? The disk 2 searches for its disk 1 and remembers when it finds it. That way only 12 games could possibly search and it would only happen once per game.

**Comment by RisingFog** [comment]
@tueidj Mind providing examples of how they're listed in Dolphin? I only have one multi-disc game which has DISC TWO listed at the end of the name.

@tony971 You still have to go through the entire list to find the match. The INI option could work (like multi-disc = 1 or something along those lines), but it's more effort for a minor UI change.

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 2]
```diff
@@ -481,7 +481,10 @@ void CGameListCtrl::InsertItemInReportView(long _Index)
 		}
 	}
 
-	SetItem(_Index, COLUMN_TITLE, StrToWxStr(name), -1);
+	if (rISOFile.IsDiscTwo() && !name.find("DISC TWO", 0))
```
Comment (JosJuice) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: Does searching for "DISC TWO" really work with game names in languages other than English?

Comment (RisingFog) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: Unsure, I only have one multi-disc game which uses DISC TWO. I'd like confirmation to clear out the majority of edge cases

**Comment by RisingFog** [comment]
bump

**Comment by JosJuice** [comment]
I still think `name.find("Disc 2", 0)` is a bit bad, but this PR should be fine anyway. It's much better than what Dolphin currently has, and I can't think of a better way to do it.

**Comment by RisingFog** [comment]
@JosJuice if there are more use cases in terms of having Disc 2 already in the title, then I will add it. I have yet to run into that though.

File: Source/Core/DolphinWX/GameListCtrl.cpp [hunk 3]
```diff
@@ -481,7 +481,10 @@ void CGameListCtrl::InsertItemInReportView(long _Index)
 		}
 	}
 
-	SetItem(_Index, COLUMN_TITLE, StrToWxStr(name), -1);
+	if (rISOFile.IsDiscTwo() && !name.find("Disc 2", 0))
```
Comment (lioncash) [inline Source/Core/DolphinWX/GameListCtrl.cpp]: `name.find("Disc 2", 0) == std::string::npos` the returned result for "not found" is the constant `-1` so comparing against zero is incorrect

**Comment by RisingFog** [comment]
bump

**Comment by shuffle2** [comment]
"I still think name.find("Disc 2") is a bit bad, but this PR should be fine anyway. It's much better than what Dolphin currently has, and I can't think of a better way to do it."

Well, the better way is probably to use regex...

**Comment by JMC47** [comment]
Any updates on this?

**Comment by RisingFog** [comment]
@JMC47 this can be merged any time

**Comment by degasus** [comment]
I don't see a reason to use regexp but to limit the position of this string and to have a wildcard for the whitespace. So I'm fine with this patch.

LGTM

**Comment by hong620** [comment]
@RisingFog 
seems problem on Japanese & Korean Text in Title List UI.
https://code.google.com/p/dolphin-emu/issues/detail?id=6852#c12

i guess it's other issues, not the issue 6852 one.
but i'm not sure.

**Comment by tapcio** [comment]
With this PR some game titles looks now like: "Resident Evil 0 DISC 2 (Disc 2)" or "RESIDENT EVIL-CODE:Veronica-X Disc2 (Disc 2)".

**Comment by RisingFog** [comment]
@tapcio Thanks for bringing that to my attention, I did not have a big enough collection of multi-disc games. Will roll out a new PR to fix the issue.

**Comment by JosJuice** [comment]
It seems like "disc2" also is used sometimes. The check should ideally be case insensitive.

**Comment by tapcio** [comment]
@JosJuice 
Yes, like "Resident Evil Disc2". Also I have notice that "MGS: The Twin Snakes Disc 2" is not affected by this PR.

**Comment by RisingFog** [comment]
@tapcio can you try PR #2467 to see if it resolves the issue?

**Pull request merged**